src/datasets/load.py

```diff
@@ -782,22 +787,21 @@ def __init__(
         download_config: Optional[DownloadConfig] = None,
         download_mode: Optional[DownloadMode] = None,
     ):
-        if data_files is None and data_dir is not None:
-            data_files = os.path.join(data_dir, "**")
 
         self.name = name
         self.data_files = data_files
-        self.downnload_config = download_config
+        self.data_dir = str(Path(data_dir).resolve()) if data_dir else str(Path().resolve())
+        self.download_config = download_config
         self.download_mode = download_mode
         increase_load_count(name, resource_type="dataset")
 
     def get_module(self) -> DatasetModule:
         patterns = (
-            sanitize_patterns(self.data_files)
-            if self.data_files is not None
-            else get_patterns_locally(str(Path().resolve()))
+            sanitize_patterns(self.data_files) if self.data_files is not None else get_patterns_locally(self.data_dir)
```

Comment:
(following previous comment)

in that case, you can check if `self.data_dir` is None here:

```suggestion
            sanitize_patterns(self.data_files)
            if self.data_files is not None
            else get_patterns_locally(self.data_dir)
            if self.data_dir is not None
            else get_patterns_locally(str(Path().resolve()))
        )
```

Reply:
[0840185](https://github.com/huggingface/datasets/pull/4144/commits/08401858ba7660d77b2f4a7bea3f8c55ef9a9931)

do you think there is a need to do `sanitize_patterns(str(Path(self.data_dir).resolve())` here or simply `sanitize_patterns(self.data_dir)` is enough, as you suggested? `sanitize_patterns(self.data_dir)` works correctly but then I don't see a need for using `str(Path().resolve()` in the next line, when self.data_dir is None.

Reply:
Both are fine, but using absolute paths with `Path(self.data_dir).resolve()` is usually good practice so feel free to add it. Relative paths sometimes cause unwanted behaviors, so let's use resolve them to absolute paths when we can